lib/libutils/ext/include/util.h

```diff
@@ -27,11 +30,20 @@
 	   __typeof__(b) _b = (b); \
 	 _a < _b ? _a : _b; }))
 #else
-#define MAX(a, b)	(((a) > (b)) ? (a) : (b))
-#define MIN(a, b)	(((a) < (b)) ? (a) : (b))
+#define MAX(a, b)	MAX_ASM(a, b)
+#define MIN(a, b)	MIN_ASM(a, b)
 #endif
 #endif
 
+/*
+ * MAX_UNSAFE/MIN_UNSAFE aim at supporting MAX/MIN with
+ * non-typed arguments in some C source file implementation
```

Comment:
I don't buy this "non-typed argument" thing. As I said, the typeof version of MIN() works fine with constants.

```
$ cat test.c
#include <stdio.h>

#define MIN(a, b) \
        (__extension__({ __typeof__(a) _a = (a); \
           __typeof__(b) _b = (b); \
         _a < _b ? _a : _b; }))

int main(int argc, char *argv[])
{
	printf("0x%x\n", MIN(0x1234, 0x5678));
	return 0;
}
$ arm-linux-gnueabihf-gcc -o test test.c
$ gcc -o test test.c && ./test
0x1234
$ 
```

Reply:
Sure, what I'm saying is we can't pretend we have understood what is happening (basically, that the macros can't be used with constants) when we have not. That's why I would remove this hunk and just define _UNSAFE above.